articles/github-actions/index.md

```diff
@@ -0,0 +1,137 @@
+
+![ga img](/engineering-education/github-actions/ga4.png)
+
+4. You can monitor you workflow logs in the same **Actions** tab after you have set it up.
+
+![log img](/engineering-education/github-actions/log.png)
+
+### How to create custom actions for common uses.
+
+#### Continuous Integration
+The whole idea behind [continuous integration](https://en.wikipedia.org/wiki/Continuous_integration) or CI is to have developers submit their code to the main codebase in small maintainable chunks, usually on a daily basis. Those changes should be automatically tested against the main codebase.
+
+You can use your own project, but I am using a simple node project.
+
+Create a new file in `.github/workflows/action.yml`.
+
+```yml
+name: Node Continuous Integration
+
+on:
+    pull_request:
+        branches: [ master ]
+jobs:
+    job_1_name:
+        runs-on: ubuntu-latest
+        steps:
+            - uses: actions/checkout@v2
+            - uses: actions/setup-node@v1
+              with:
+                node-version: 12
+            - run: npm ci
+            - run: npm test
+            - run: npm run build
+
+```
+
+The above action does the following.
+
+1. We set up an action with name *Node Continuous Integration*.
+2. This action is triggered when there is a *pull request on the master branch*. The event which triggers the action is mentioned under `on:`
+3. An action has one or more `jobs`. The first does  the following
+ * Spins up an ubuntu virtual machine.
+ * `actions/checkout@v2` Brings the source code to the current working directory. `checkout@v2` is an existing action from GitHub.
+ * Installs `node`.
+ * Install app dependencies, using `npm ci`.
+ * Runs the command `npm test` followed by `npm run build`, we can use our preferred libraries like *jest* and *webpack*, it executes the test *script* mentioned in `package.json`.
+
+If there are any errors or build failures, then it shows up in the log in the actions tab of our repository by a green check or a red cross.
+
+
+![greencheck img](/engineering-education/github-actions/greencheck.png)
+
+### Continuous Deployment
+Once you have a valid pull request and when we merge that code into the master branch, we also want to deploy the app to our customers. [Continuous deployment](https://www.atlassian.com/continuous-delivery/continuous-deployment) is about pushing that code out to the customer.
+If your website is hosted on Github, then you can use the `github-pages` action. However, if a third-party provider hosts your app, then they provide an [SFTP](https://www.digitalocean.com/community/tutorials/how-to-use-sftp-to-securely-transfer-files-with-a-remote-server) option to upload the site data. This is fine when we are doing it manually but not so easy when we are using a CD tool. We will have to share a secret token with GitHub. I will show how to deploy to [firebase](http://firebase.google.com/).
```

Comment:
It might be good to clarify that using a SFTP option is only one of many options for how someone might be deploying their app. Often there may be a more custom system in place or they may just be doing something like using docker images. Regardless, nearly all of these will require a 'secret' of some sort which is what this section addresses. It might be good to clarify that this is how to handle secrets in general with Github actions, not just in the SFTP use case.

Reply:
Added SSH and FTP, with their links to github actions.